solana/pyth2wormhole/client/src/lib.rs

```diff
@@ -50,19 +51,24 @@ use pyth2wormhole::{
     initialize::InitializeAccounts,
     set_config::SetConfigAccounts,
     AttestData,
-    Pyth2WormholeConfig,
 };
 
+pub use pyth2wormhole::Pyth2WormholeConfig;
+
 pub use attestation_cfg::{
     AttestationConditions,
     AttestationConfig,
     P2WSymbol,
 };
-pub use batch_state::{
-    BatchState,
-    BatchTxStatus,
+pub use batch_state::BatchState;
+pub use util::{
+    RLMutex,
+    RLMutexGuard,
 };
 
+/// Future-friendly version of solitaire::ErrBox
+pub type ErrBoxSend = Box<dyn std::error::Error + Send + Sync>;
```

Comment:
Have you seen anyhow (https://docs.rs/anyhow/latest/anyhow/)? It allows you to easily propagate any error which implements `std::error::Error` by converting it to an `anyhow::Error`. You can then return `Result<T, anyhow::Error>` from your functions, and across threads, without having to do manual error conversion using `map_err`. It makes dealing with errors in Rust much nicer.

Reply:
I've heard of it, but trading this one-liner for a whole dependency did not seem justified. I checked and we have 3-4 trivial map_err's in the whole pyth2wormhole cargo workspace. I'm sure this could be a good step to take in the future